### ISSUE TITLE
Implement GraphQL Connections model

### ISSUE THREAD

**Issue by johnrom** [OrchardCMS/OrchardCore#5693]
https://graphql.org/learn/pagination/#complete-connection-model

Related: #4291, did some basic pagination but did not implement Connections.

GraphQL Connections are considered a best practice for GraphQL pagination, and without it developers do not get the full benefits of graphql (partial queries of related items in sequential pagination, etc).

**Comment by carlwoodhouse** [comment]
Agreed we should implement connections (and said as much in the original issue) we probably need to support both models now though so we dont break everyones existing apps.
